pkg/network/encoding/encoding.go

```diff
@@ -50,9 +51,11 @@ func modelConnections(conns *network.Connections) *model.Connections {
 	agentConns := make([]*model.Connection, len(conns.Conns))
 	domainSet := make(map[string]int)
 	routeIndex := make(map[string]RouteIdx)
+	httpIndex := FormatHTTPStats(conns.HTTP)
 
 	for i, conn := range conns.Conns {
-		agentConns[i] = FormatConnection(conn, domainSet, routeIndex)
+		httpKey := http.NewKey(conn.Source, conn.Dest, conn.SPort, conn.DPort, "")
+		agentConns[i] = FormatConnection(conn, domainSet, routeIndex, httpIndex[httpKey])
```

Comment:
That's a good point. I _think_ it's okay to leave the code as it is because we definitely don't want to "bind" the same stats to multiple connections which would result in counting the same thing multiple times. 

On the other hand, it means that we might be misrepresenting which connections are responsible for which stats.
The obvious fix I guess would be adding PID to the HTTP key (for the record we don't consider PID in DNS either), but I can't think of a realistic scenario where this would be an issue.

One example I can think of is multiple containers in the same kubernetes pod. They share the same network namespace, so they would have the same IP. But I guess the only possible example of tuple collision would be in the case where multiple containers end up using the same source port number at different times within a 30s window which I think it's a pretty contrived scenario.

The other example I guess is forked processes using the same socket, but in that case I guess we don't care about PID resolution since they will be aggregated anyways?

Reply:
Since these metrics contain counts, we'll wind up double counting in the forked process case. 


Could we get around this by adding a `delete(httpIndex, httpKey)` at each iteration?

Reply:
oh, you're right! yes, let's do that